plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -240,8 +247,22 @@ public ClickHouseClient(
         this.uuidType = typeManager.getType(new TypeSignature(StandardTypes.UUID));
         this.ipAddressType = typeManager.getType(new TypeSignature(StandardTypes.IPADDRESS));
         JdbcTypeHandle bigintTypeHandle = new JdbcTypeHandle(Types.BIGINT, Optional.of("bigint"), Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
-        this.connectorExpressionRewriter = JdbcConnectorExpressionRewriterBuilder.newBuilder()
+        JdbcConnectorExpressionRewriterBuilder connectorExpressionRewriterForJoinBuilder = JdbcConnectorExpressionRewriterBuilder.newBuilder()
                 .addStandardRules(this::quoted)
+                .map("$equal(left, right)").to("left = right")
```

Comment:
Should we also update the TupleDomain based implementation which currently treats Varchar as VarbinaryMapping which restricts pushdown ?

Reply:
I think it's a separate, not related effort. 
Discussed here: https://github.com/trinodb/trino/pull/23455#discussion_r1764838133